modules/inventory/server.lua

```diff
@@ -751,6 +751,17 @@ function Inventory.AddItem(inv, item, count, metadata, slot, cb)
 
 			local toSlot, slotMetadata, slotCount
 
+			if metadata and metadata.imageurl then
+				local isValidURL = IsValidImageUrl(metadata.imageurl)
+				if not isValidURL then
+					response = 'invalid_url'
+					DiscordLog("AddItem Failure", ('"%s" attempted to add item with invalid url "%s"\nMetadata=`%s`'):format(GetPlayerName(inv.id), metadata.imageurl, json.encode(metadata)), metadata.imageurl, 16711680)
+					if cb then return cb(success, response) end
+					return success, response
+				end
+				DiscordLog("AddItem", ('"%s" added %sx %s to "%s" with the image url "%s"\nMetadata=`%s`'):format(GetPlayerName(inv.id), count, item.name, metadata.label or inv.label, metadata.imageurl, json.encode(metadata)), metadata.imageurl, 65280)
+			end
+
```

Comment:
This all needs to be part of the `Items.Metadata` function, not `Inventory.AddItem`.
Can also do `if IsValidImageUrl(...) then`, and formatting generally needs improvement
- empty line before and after `if` statement (if there's other code in the scope)
- `'...'` for strings like everywhere else
- "AddItem" and "AddItem Failure" is a weird title to give the embed.